nshutijonathan/Auto-mart3
# #166784828 Build out swagger for Auto-mart REST endpoints

#### What does this PR do?
- Swagger UI Implementation
#### Description of Task to be completed?
- Added tokens authorization on swagger
- Removed console.log in source code
- Refactored codebase to meet standards
#### Have The following endpoint working and validated and tested in P
* POST/localhost:3000/api/v2/auth/signup
* POST/localhost:3000/api/v2/auth/signin
* POST/localhost:3000/api/v2/auth/signup/admin
* GET/localhost:3000/api/v2/users
*   GET/localhost:3000/api/v2/users/<:id>
*   DELETE/localhost:3000/api/v2/users/<:id>
*   POST/localhost:3000/api/v2/car
*   GET/localhost:3000/api/v2/cars
*   GET/localhost:3000/api/v2/users/me
*  PUT/localhost:3000/api/v2/<:email>/reset_password
*   POST/localhost:3000/api/v2/order
* PATCH/localhost:3000/api/v2/<:order-id/order
*  PATCH/localhost:3000/api/v2/car/<:car-id>/status
* PATCH/locahost:3000/api/v2/car/<:car-d>/price
* GET / locahost:3000/api/v2/car/<:id>
* GET / locahost:3000/api/v2/car?status=available
#### How should this be manually tested?
* Download or clone the tutorial project code from [this repo](https://github.com/nshutijonathan/Auto-Mart3), Install all required NPM  packages by running NPM install from the command line in the project root folder (where the package.json is located).
* Start the API test by running NPM test from the command line in the project root 
* Start the API by running NPM  start from the command line in the project root folder, you should see the message Server listening on port 3000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
* Start API documentation from here [Auto-mart-swagger](https://auto-mart-challenge3.herokuapp.com/apis-docs/)
* You must have node environment configured  and .env file in the root folder
* to test authentication and authorization ,add this to the header 
Key: :x-auth-token,value:JWT TOKEN
#### Any background context you want to provide?
* There is Implementation of JavaScript(ES6) logic with SQL database to persist some data
#### What are the relevant pivotal tracker stories?
[#166784828](https://www.pivotaltracker.com/n/projects/2346053/stories/166784828 )

#### Screenshots
![swagger](https://user-images.githubusercontent.com/16412410/59899387-cd77e200-93f4-11e9-8c51-c9f0bfb7b76a.PNG)
